### PR TITLE
Pat 2155 fix viper processing oom errors

### PR DESCRIPTION
**Pull request by garethmdavies**
To avoid OOM errors during the load of the viper file  (200Mb+) from S3 into memory, the processing has been changed so that as the file is streamed into memory only the latest viper offender data is loaded with historical records skipped. This process **could** be extended further to extract just the columns of data needed by the risk profiler.

To test this new functionality, the tests will now use s3 in embedded localstack instead of the filesystem to process the risk-profiler csv files.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/services/S3FileService.kt [hunk 1]
```diff
@@ -70,20 +61,38 @@ class S3FileService(
...
+  private fun getViperFile(s3ObjectContent: S3ObjectInputStream): ByteArray {
+    val reader = BufferedReader(InputStreamReader(s3ObjectContent))
+    var row: List<String>
+    val rowList = mutableListOf<String>()
+
+    while (true) {
+      val line = reader.readLine() ?: break
+      row = line.split(",")
+      if (row[Viper.RECORD_ID] != "0") {
+        rowList.add(row.joinToString(","))
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/services/S3FileService.kt:73]: minor, but cant you just add(line) ?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/services/FileService.kt [hunk 2]
```diff
@@ -1,6 +1,8 @@
 package uk.gov.justice.digital.hmpps.riskprofiler.services
 
+import uk.gov.justice.digital.hmpps.riskprofiler.datasourcemodel.FileType
+
 interface FileService {
-  fun getLatestFile(fileLocation: String): PendingFile?
+  fun getLatestFile(fileLocation: String, fileType: FileType?): PendingFile?
   fun deleteHistoricalFiles(fileLocation: String)
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/services/FileService.kt:7]: Possibly for another time, but can we get rid of this interface and the 'DefaultFileService' entirely now? I guess this pre-dated the convention of using localstack locally

File: src/test/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/services/S3FileServiceTest.kt [hunk 3]
```diff
@@ -29,9 +29,7 @@ class S3FileServiceTest {
   @Before
   fun setup() {
     service = S3FileService(
-      amazonS3Client,
-      amazonS3ViperClient,
-      ImmutableList.of("risk-profiler|s3Client", "viper-risk-profiler|viperS3Client")
+      amazonS3Client
     )
   }
```
Comment (steverendell) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/services/S3FileServiceTest.kt:34]: Probably could do with a test that exercises getViperFile()

**Review by steverendell — DISMISSED** [review]
lgtm with comments

File: src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/services/FileService.kt (hunk 2 above)
Comment (garethmdavies) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/riskprofiler/services/FileService.kt:7]: I think so. Just didn't want to make too many changes in one PR.

**Review by steverendell — APPROVED** [review]
lgtm

**Pull request merged**